Client/Frontend/Fakespot/Views/FakespotMessageCardView.swift

```diff
@@ -0,0 +1,275 @@
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+import Common
+import ComponentLibrary
+
+struct FakespotMessageCardViewModel {
+    var title: String
+    var description: String?
+    var linkText: String?
+    var primaryActionText: String?
+
+    var a11yCardIdentifier: String
+    var a11yTitleIdentifier: String
+    var a11yDescriptionIdentifier: String?
+    var a11yPrimaryActionIdentifier: String?
+    var a11yLinkActionIdentifier: String?
+}
+
+final class FakespotMessageCardView: UIView, ThemeApplicable, Notifiable {
+    enum CardType: String, CaseIterable, Identifiable {
+        case confirmation
+
+        var id: String { self.rawValue }
+
+        func primaryButtonText(theme: Theme) -> UIColor {
+            switch self {
+            case .confirmation: return theme.colors.textPrimary
+            }
+        }
+
+        func primaryButtonBackground(theme: Theme) -> UIColor {
+            switch self {
+            case .confirmation: return theme.colors.actionConfirmation
+            }
+        }
+
+        func cardBackground(theme: Theme) -> UIColor {
+            switch self {
+            case .confirmation: return theme.colors.layerConfirmation
+            }
+        }
+
+        var iconImageName: String {
+            switch self {
+            case .confirmation: return StandardImageIdentifiers.Large.checkmark
+            }
+        }
+    }
+
+    private enum UX {
+        static let linkFontSize: CGFloat = 12
+        static let buttonFontSize: CGFloat = 16
+        static let buttonVerticalInset: CGFloat = 12
+        static let buttonHorizontalInset: CGFloat = 16
+        static let buttonCornerRadius: CGFloat = 13
+        static let contentHorizontalSpacing: CGFloat = 4
+        static let contentVerticalSpacing: CGFloat = 8
+        static let iconStackViewSpacing: CGFloat = 4
+        static let horizontalStackViewSpacing: CGFloat = 12
+        static let verticalStackViewSpacing: CGFloat = 4
+        static let iconSize: CGFloat = 24
+        static let iconMaxSize: CGFloat = 58
+        static let titleFontSize: CGFloat = 13
+        static let descriptionFontSize: CGFloat = 13
+    }
+
+    private lazy var cardView: CardView = .build()
+    private lazy var contentView: UIView = .build()
+
+    private lazy var iconStackView: UIStackView = .build { stackView in
+        stackView.axis = .vertical
+        stackView.distribution = .equalCentering
+        stackView.spacing = UX.iconStackViewSpacing
+    }
+
+    private lazy var containerStackView: UIStackView = .build { stackView in
+        stackView.axis = .vertical
+        stackView.spacing = UX.horizontalStackViewSpacing
+    }
+
+    private lazy var infoContainerStackView: UIStackView = .build { stackView in
+        stackView.axis = .horizontal
+        stackView.spacing = UX.horizontalStackViewSpacing
+    }
+
+    private lazy var labelContainerStackView: UIStackView = .build { stackView in
```

Comment:
Which ones do you think should be removed? 
Imo, most of them are there for a reason. 
The infoContainerStackView is there to keep the icon and labels aligned horizontally as they grow with accessibility. Spacers help the icon to be centered with the title as is grows with the accessibility size.
The label container helps me remove from superview without remaking all the constraints. Remember some elements will not be there and config is called after the init (after we add them with the layoutViews())
This happens for container stackview as well, primary button can be absent so it also needs to be removed without breaking the layout.

 I'm open for suggestions.

Reply:
Just curious, what's the concern with stackviews and spacer views?

Reply:
There were situations previously were stack views had let's say unexpected layout behaviours and Winnies suggests we should minimise their use to avoid these potentials problems. In my experience with them they can definitely impose some problems but imo the benefits out-weight the issues.
I find them really useful in a dinamic context with accesibility and growing componet sizes. Also removing and adding components without remaking constraints. And another plus they prepare us for the eventual transition to SwiftUI where the UI is only done by stacks.

Reply:
@razvanlitianu only the `containerStackView` and `labelContainerStackView` are really necessary. Happy to show you how I'd do it.

@OrlaM stackviews tend to cause issues with the layout - especially with dynamic type (we ran into this in multiple places already). Spacer views itself are not an issue, they just make it harder to see at first glace what the spacing between elements is and adjust it. I find it harder to work with them especially when space needs to be updated when design reviews the layout.